Neos.FluidAdaptor/Classes/Core/Widget/AjaxWidgetComponent.php

```diff
@@ -75,15 +76,15 @@ public function handle(ComponentContext $componentContext)
      * If the request contains an argument "__widgetId" the context is fetched from the session (AjaxWidgetContextHolder).
      * Otherwise the argument "__widgetContext" is expected to contain the serialized WidgetContext (protected by a HMAC suffix)
      *
-     * @param Request $httpRequest
+     * @param ServerRequestInterface $httpRequest
      * @return WidgetContext
      */
-    protected function extractWidgetContext(Request $httpRequest)
+    protected function extractWidgetContext(ServerRequestInterface $httpRequest)
     {
-        if ($httpRequest->hasArgument('__widgetId')) {
-            return $this->ajaxWidgetContextHolder->get($httpRequest->getArgument('__widgetId'));
-        } elseif ($httpRequest->hasArgument('__widgetContext')) {
-            $serializedWidgetContextWithHmac = $httpRequest->getArgument('__widgetContext');
+        if (isset($httpRequest->getQueryParams()['__widgetId'])) {
```

Comment:
`QueryParams` should be fine instead of the unified arguments, or is there a use-case to submit the `__widget*` arguments via POST params?

Reply:
you never know, but I can imagine that might happen...

Reply:
Yep yep. So we need the ArgumentsHelper mumbo-jumbo here. Or maybe we can decouple this Fluid Widget stuff from Http and only depend on ActionRequest?